src/pkgagent/agent/pkgagent.c

```diff
@@ -458,7 +461,8 @@ void ReadHeaderInfo(Header header, struct rpmpkginfo *pi)
         case RPMTAG_BUILDTIME:
           t = atol(msgstr);
           tp = &t;
-          strncpy(pi->buildDate,trim(asctime(gmtime((time_t*)tp))),sizeof(pi->buildDate));
+          strncpy(pi->buildDate,trim(asctime(gmtime((time_t*)tp))),sizeof(pi->buildDate)-1);
+	  pi->buildDate[sizeof(pi->buildDate)-1] = '\0';
```

Comment:
Please fix the indentation. Please use spaces instead of tabs.

Reply:
Sorry, corrected.